### ISSUE TITLE
Warning about "is not" comparison

### ISSUE THREAD

**Issue by arokem** [yeatmanlab/pyAFQ#388]
I am seeing this one:

```
/Users/arokem/source/pyAFQ/AFQ/api.py:402: SyntaxWarning: "is not" with a literal. Did you mean "!="?
```

Need to go check what this is all about.

**Comment by 36000** [comment]
My bad. That check is not even relevant with our current API anyways. Fixed in #389. I wonder if there is a way to have argument checking performed automatically from the docstring. Because AFQ.api should probably be argument checking (in this case, brain_mask should never be a string).

**Comment by arokem** [comment]
I am not 100% sure, but I think that judicious use of [attrs](https://www.attrs.org/en/stable/) with [type annotation](https://www.attrs.org/en/stable/types.html) might help with this kind of checking, avoiding boilerplate. Worth taking a look?
